### PR TITLE
[WIP] Separated generators from the main library

### PR DESCRIPTION
**Pull request by effectfully**
I couldn't put generators in their own library, because they depend on `language-plutus-core` and tests of the latter depend on generators, so there is a loop.

This also fixes an accident TODO: replaces `fmap BSL.fromStrict . Gen.bytes` by `genLowerBytes`.

### CODE REVIEW

**Comment by vmchale** [comment]
Would you consider making a `language-plutus-core-tests` library as well? Or was that considered already and we decided against it?

**Comment by effectfully** [comment]
> Would you consider making a `language-plutus-core-tests` library as well?

I'd prefer to keep tests within libraries they test.

> Or was that considered already and we decided against it?

I think, no. @mchakravarty, @michaelpj ^

File: language-plutus-core/generators/Language/PlutusCore/Generators/Interesting.hs [hunk 1]
```diff
@@ -0,0 +1 @@
+module Language.PlutusCore.Generators.Interesting where
```
[the remark this answers is not in the document]
Comment (effectfully) [reply language-plutus-core/generators/Language/PlutusCore/Generators/Interesting.hs]: I wanted to move terms from the tests to this location. But it turned not to be that straightforward, so I'll do this in a separate PR.

**Comment by michaelpj** [comment]
@effectfully I'll give them a little while to look at this, then I'll force-merge it if it's still stuck (I've looked at the jobset and the jobs have in fact succeeded).

**Comment by michaelpj** [comment]
Okay, merging.

**Pull request merged**